From 32382f01bb65cc81acb835b79ef6f64935bc9524 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@luon.net>
Date: Sun, 25 Nov 2012 21:10:00 +0100
Subject: [PATCH 1/2] Switch the default resample quality to 1

Swithc the standard quality level of the speex resampler down a notch
from -3 to -1 to lower cpu usage by the resampler. This has been
dicussed for a while, without real oposition and has been the default in
at least Ubuntu for a while.
---
 man/pulse-daemon.conf.5.xml.in |    2 +-
 src/daemon/daemon.conf.in      |    2 +-
 src/pulsecore/core.c           |    2 +-
 src/pulsecore/resampler.c      |    6 +++---
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/man/pulse-daemon.conf.5.xml.in b/man/pulse-daemon.conf.5.xml.in
index c270d2b..43257a6 100644
--- a/man/pulse-daemon.conf.5.xml.in
+++ b/man/pulse-daemon.conf.5.xml.in
@@ -100,7 +100,7 @@ USA.
       desktop CPUs the float point resampler is a lot faster, and it
       also offers slightly better quality. See the output of
       <opt>dump-resample-methods</opt> for a complete list of all
-      available resamplers. Defaults to <opt>speex-float-3</opt>. The
+      available resamplers. Defaults to <opt>speex-float-1</opt>. The
       <opt>--resample-method</opt> command line option takes precedence.
       Note that some modules overwrite or allow overwriting of the
       resampler to use.</p>
diff --git a/src/daemon/daemon.conf.in b/src/daemon/daemon.conf.in
index dff97ae..2089058 100644
--- a/src/daemon/daemon.conf.in
+++ b/src/daemon/daemon.conf.in
@@ -54,7 +54,7 @@ ifelse(@HAVE_DBUS@, 1, [dnl
 ; log-time = no
 ; log-backtrace = 0
 
-; resample-method = speex-float-3
+; resample-method = speex-float-1
 ; enable-remixing = yes
 ; enable-lfe-remixing = no
 
diff --git a/src/pulsecore/core.c b/src/pulsecore/core.c
index e4f9140..a9f1b7d 100644
--- a/src/pulsecore/core.c
+++ b/src/pulsecore/core.c
@@ -141,7 +141,7 @@ pa_core* pa_core_new(pa_mainloop_api *m, pa_bool_t shared, size_t shm_size) {
     c->disable_remixing = FALSE;
     c->disable_lfe_remixing = FALSE;
     c->deferred_volume = TRUE;
-    c->resample_method = PA_RESAMPLER_SPEEX_FLOAT_BASE + 3;
+    c->resample_method = PA_RESAMPLER_SPEEX_FLOAT_BASE + 1;
 
     for (j = 0; j < PA_CORE_HOOK_MAX; j++)
         pa_hook_init(&c->hooks[j], c);
diff --git a/src/pulsecore/resampler.c b/src/pulsecore/resampler.c
index a13e700..5f8e9bd 100644
--- a/src/pulsecore/resampler.c
+++ b/src/pulsecore/resampler.c
@@ -235,7 +235,7 @@ pa_resampler* pa_resampler_new(
 
     if (method == PA_RESAMPLER_AUTO) {
 #ifdef HAVE_SPEEX
-        method = PA_RESAMPLER_SPEEX_FLOAT_BASE + 3;
+        method = PA_RESAMPLER_SPEEX_FLOAT_BASE + 1;
 #else
         if (flags & PA_RESAMPLER_VARIABLE_RATE)
             method = PA_RESAMPLER_TRIVIAL;
@@ -552,10 +552,10 @@ pa_resample_method_t pa_parse_resample_method(const char *string) {
             return m;
 
     if (pa_streq(string, "speex-fixed"))
-        return PA_RESAMPLER_SPEEX_FIXED_BASE + 3;
+        return PA_RESAMPLER_SPEEX_FIXED_BASE + 1;
 
     if (pa_streq(string, "speex-float"))
-        return PA_RESAMPLER_SPEEX_FLOAT_BASE + 3;
+        return PA_RESAMPLER_SPEEX_FLOAT_BASE + 1;
 
     return PA_RESAMPLER_INVALID;
 }
-- 
1.7.10.4

